drivers/ethernet/eth_tsn_nic.c

```diff
@@ -136,20 +179,40 @@ struct eth_tsn_nic_data {
 	struct dma_tsn_nic_desc tx_desc;
 	struct dma_tsn_nic_desc rx_desc;
 
-	/* TODO: Maybe this needs to be allocated dynamically */
+	/* TODO: Maybe these need to be allocated dynamically */
+	struct tx_buffer tx_buffer;
 	uint8_t rx_buffer[BUFFER_SIZE];
 
 	struct dma_tsn_nic_result res;
 
+	struct k_work tx_work;
 	struct k_work rx_work;
 
 	bool has_pkt; /* TODO: This is for test only */
 };
 
-static void eth_tsn_nic_isr(const struct device *port)
+static void eth_tsn_nic_isr(const struct device *dev)
 {
 	/* TODO: Implement interrupts */
-	ARG_UNUSED(port);
+	ARG_UNUSED(dev);
+}
+
+static void tx_desc_set(struct dma_tsn_nic_desc *desc, uintptr_t addr, uint32_t len)
+{
+	uint32_t control_field;
+	uint32_t control;
+
+	desc->control = sys_cpu_to_le32(DESC_MAGIC);
+	control_field = DESC_STOPPED;
+	control_field |= DESC_EOP;
+	control_field |= DESC_COMPLETED;
+	control = sys_le32_to_cpu(desc->control & ~(LS_BYTE_MASK));
+	control |= control_field;
+	desc->control = sys_cpu_to_le32(control);
```

Comment:
`desc->control`을 건드렸다가 `control_field`를 건드렸다가 왔다갔다 하네요
```suggestion
	desc->control = sys_cpu_to_le32(DESC_MAGIC);
	control = sys_le32_to_cpu(desc->control & ~(LS_BYTE_MASK));
	control |= DESC_STOPPED;
	control |= DESC_EOP;
	control |= DESC_COMPLETED;

	desc->control = sys_cpu_to_le32(control);
```

Reply:
이건 그대로 가져다 썼는데 지금 보니 이상하게 꼬여 있네요.. 앞으로는 이미 있는 코드여도 너무 믿지 않고 다시 보도록 하겠습니다..